core/src/apps/management/recovery_device/homescreen.py

```diff
@@ -63,7 +73,10 @@
             continue
 
         try:
-            secret, word_count, backup_type = await _process_words(ctx, words)
+            secret, backup_type = await _process_words(ctx, words)
+            # If _process_words succeeded, backup_type will be set.
+            # Otherwise we are still in "first step".
```

Comment:
Maybe we should clarify here that the `backup_type` gets set **only** if `_process_words` succeeds. Otherwise it raises an exception, the code below is not executed and `is_first_step` stays `True`.

In another words, we could actually change the second line to:
```
is_first_step = False
```
but that might be even more confusing, so maybe let's just elaborate with a comment? What do you think?

Reply:
Adding comment that _process_words raises an exception in case it fails should be enough IMHO.

Reply:
so basically what you're saying is, we could make the code clearer by changing to `is_first_step = False` *and* removing the comment :)
(then the try-except flow clearly indicates that an exception might be happening somewhere around here)

Reply:
Yes. I'm fine that.

Reply:
(but of course the comment is a good place to explain that at this point we definitely have both backup_type and word_count and so the original invariants are satisfied)